### PR TITLE
Adding testgrid content for sig-storage container object storage project

### PR DESCRIPTION
**Pull request by brahmaroutu**
This PR will enable prow for one of the 4 repos new created for the project implementing container-object-storage-interface under sig-storage. 
The PR is adding a CI and test grid dashboard for container-object-storage-interface-api repository.
Similar configuration will follow for container-object-storage-interface-controller and container-object-storage-interface-provisioner-sidecar and container-object-storage-interface-csi-adapter.

/sig-storage
ping @xing-yang @wlan0 
/release-note-none

### CODE REVIEW

File: config/jobs/kubernetes-sigs/container-object-storage-interface/container-object-storage-interface-api.yaml [hunk 1]
```diff
@@ -0,0 +1,32 @@
+presubmits:
+  kubernetes-sigs/container-object-storage-interface-api:
+  - name: pull-container-object-storage-iinterface-api-build
```
Comment (wlan0) [inline config/jobs/kubernetes-sigs/container-object-storage-interface/container-object-storage-interface-api.yaml:3]: typo

**Comment by wlan0** [comment]
LGTM

File: config/jobs/kubernetes-sigs/container-object-storage-interface/container-object-storage-interface-api.yaml (hunk 1 above)
Comment (brahmaroutu) [reply config/jobs/kubernetes-sigs/container-object-storage-interface/container-object-storage-interface-api.yaml]: fixed, thanks for the review.

**Comment by xing-yang** [comment]
Looks good to me.  @msau42 can you also take a look?

/assign @msau42

**Pull request merged**